app/javascript/controllers/tab_controller.js

```diff
@@ -0,0 +1,33 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  switch() {
+    this.#deactivate(this.#currentlyActiveButton())
+    this.#makeActive(this.element)
+  }
+
+  #currentlyActiveButton() {
+    return this.#tabList.querySelector('[role=tab].active')
+  }
+
+  get #tabList() {
+    return this.element.closest('[role=tablist]')
+  }
+
+  #makeActive(button) {
+    button.classList.add("active")
+    button.setAttribute('aria-selected', true)
+    this.#tabPanelFor(button).hidden = false
+  }
+
+  #deactivate(button) {
+    button.classList.remove("active")
+    button.setAttribute('aria-selected', false)
+    this.#tabPanelFor(button).hidden = true
+  }
+
+  #tabPanelFor(button) {
```

Comment:
Can you say more about all the `#` characters in the function names?

Reply:
That's how javascript denotes private methods: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties

Reply:
@jcoyne I'm not sure I've seen this before in any of our codebases. Cool!